File: fizzbuzz/fizzbuzz.rb
# ruby -r ./fizzbuzz.rb -e test

# @param {Integer} n
# @return {String[]}
def fizz_buzz(n)
    (1..n).map do
        case 
        when _1 % 3 == 0 && _1 % 5 == 0
            "FizzBuzz"
        when _1 % 3 == 0
            "Fizz"
        when _1 % 5 == 0
            "Buzz"
        else
            "#{_1}"
        end
    end
end

def test
  tests = [
    {    
      expect: ["1","2","Fizz"],
      given: 3,
    },

    {  
      expect: ["1","2","Fizz","4","Buzz"],
      given: 5,
    },

    {  
      expect: ["1","2","Fizz","4","Buzz","Fizz","7","8","Fizz","Buzz","11","Fizz","13","14","FizzBuzz"],
      given: 15,
    },
  ]

  tests.each do |test|
    given = test[:given]
    expect = test[:expect]
    res = fizz_buzz(given)
    if expect != res
      raise StandardError.new("Unexpected result, want=#{expect}, got=#{res}")
    end
  end
end

File: fizzbuzz/fizzbuzz.py
# uv run fizzbuzz.py

# /// script
# requires-python = ">=3.13"
# dependencies = []
# ///


def fizzbuzz(n):
    out = []
    for i in range(1, n+1):
        if i % 3 == 0 and i % 5 == 0:
            out.append("FizzBuzz")
        elif i % 3 == 0:
            out.append("Fizz")
        elif i % 5 == 0:
            out.append("Buzz")
        else:
            out.append(str(i))

    return out

def test():
    tests = [
        [3, ["1","2","Fizz"]],
        [5, ["1","2","Fizz","4","Buzz"]],
        [15, ["1","2","Fizz","4","Buzz","Fizz","7","8","Fizz","Buzz","11","Fizz","13","14","FizzBuzz"]],
    ]

    for n, exp in tests:
        res = fizzbuzz(n)
        if exp != res:
            raise RuntimeError(f"Unexpected result for n={n}, want={exp}, got={res}")


if __name__ == "__main__":
    test()
